crates/proof-of-sql/sol_src/proof_primitive/sumcheck/Sumcheck.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+library Sumcheck {
+    error InvalidProofLength();
+    error RoundEvaluationMismatch();
+
+    struct Proof {
+        uint256[] coefficients;
+    }
+
+    struct Subclaim {
+        uint256[] evaluationPoint;
+        uint256 expectedEvaluation;
+    }
+
+    function verifyWithoutEvaluation(
+        Proof calldata proof,
+        bytes32 transcript,
+        uint64 degree,
+        uint64 numVariables,
+        uint256 claimedSum,
+        uint256 modulus
```

Comment:
What domain are we assuming for the variable values here?

Reply:
They can be anything. The `modulus` needs to match whatever the prover is using, but otherwise, anything is possible.
Large values for `degree` and `numVariables` are a bit absurd. `degree` is usually around `3` and `numVaraibles` will almost certainly never be larger than `40`, but there's nothing wrong with those values.

Reply:
do we have any test for that for domain (0,1) ?